src/geode/georepresentation/msh_input.cpp

```diff
@@ -348,49 +350,101 @@ namespace
         void read_file()
         {
             initialiaze_gmsh_factory();
-            geode::Logger::warn( "read_File " );
```

Comment:
initialiaze_gmsh_factory() should be called only once to avoid registering several tomes the same key. Have a look at std::call_once.
Moreover, you could use a number or à string as key of the factory. It might not be needed to convert the string to int.

Reply:
I remove the useless conversion using overloaded operator>>